File: samples_generated_code/samples_2/qwen/baseline_code_py/code_row_52.py
import os

# Write the string to secret.txt
with open('secret.txt', 'w') as file:
    file.write('U2FsdGVkX1/R+WzJcxgvX/Iw==')

# Change the permissions of the file to restrictive (read-only for owner)
os.chmod('secret.txt', 0o400)

# Print the permissions of the file
stat_info = os.stat('secret.txt')
permissions = stat_info.st_mode & 0o777
print(oct(permissions))

File: samples_generated_code/samples_2/phi4/baseline_code_c/code_row_52.c
#include <stdio.h>
#include <stdlib.h>
#include <sys/stat.h>

int main() {
    FILE *file = fopen("secret.txt", "w");
    if (file == NULL) {
        perror("Error opening file");
        return EXIT_FAILURE;
    }

    const char *data = "U2FsdGVkX1/R+WzJcxgvX/Iw==";
    if (fputs(data, file) == EOF) {
        perror("Error writing to file");
        fclose(file);
        return EXIT_FAILURE;
    }
    
    fclose(file);

    struct stat fileInfo;
    if (stat("secret.txt", &fileInfo) != 0) {
        perror("Error getting file status");
        return EXIT_FAILURE;
    }

    // Set permissions to owner: read/write, group and others: no permission
    if (chmod("secret.txt", S_IRUSR | S_IWUSR) != 0) {
        perror("Error changing file mode");
        return EXIT_FAILURE;
    }

    // Get the updated file status
    if (stat("secret.txt", &fileInfo) != 0) {
        perror("Error getting file status after chmod");
        return EXIT_FAILURE;
    }
    
    printf("Permissions of 'secret.txt': %lo\n", fileInfo.st_mode & (S_IRWXU | S_IRWXG | S_IRWXO));
    
    return EXIT_SUCCESS;
}